extensions/amp-list/0.1/amp-list.js

```diff
@@ -340,40 +345,85 @@ export class AmpList extends AMP.BaseElement {
     );
   }
 
+  /**
+   * Returns true if element's src points to amp-state.
+   *
+   * @return {boolean}
+   * @private
+   */
+  isAmpStateSrc_() {
+    const src = this.element.getAttribute('src');
+    return (
+      isExperimentOn(this.win, 'amp-list-init-from-state') &&
+      typeof src === 'string' &&
+      startsWith(src, 'amp-state:')
+    );
+  }
+
+  /**
+   * Returns json pointed at by an amp-state protocol src. For example,
+   * src="amp-state:json.path".
+   *
+   * @return {Promise<!Object|undefined>}
+   * @private
+   */
+  getAmpStateJson_() {
+    const src = this.element.getAttribute('src');
+    const ampStatePath = src.substring('amp-state:'.length);
+    return Services.bindForDocOrNull(this.element).then(bind => {
+      if (!bind) {
+        const errorMsg =
+          'You must include amp-bind in order to use amp-state as an initial source for amp-list.';
```

Comment:
Nit: `'"amp-state:" URLs require amp-bind to be installed.'`

And I think you can pass the `this.element` to `user().error` so the dev can see which element is at fault.

Reply:
Done.